egs/hub4_english/s5/local/train_lm.sh

```diff
@@ -0,0 +1,250 @@
+    awk '{for (i=0; i<$1; i++) {print $0;} }' | cut -d ' ' -f 2- > \
+    ${dir}/data/text/dev.txt 
+  tail -n +$[num_dev_sentences+1] < ${dir}/train_bn96_text_with_count | \
+    awk '{for (i=0; i<$1; i++) {print $0;} }' | cut -d ' ' -f 2- > \
+    ${dir}/data/text/train_bn96.txt
+
+  # Get text from NA News corpus 
+  for x in data/local/data/na_news/*; do
+    y=`basename $x`
+    [ -f $x/corpus.gz ] && ln -sf `readlink -f $x/corpus.gz` ${dir}/data/text/${y}.txt.gz
+  done
+
+  # Get text from 1996 CSR HUB4 LM corpus
+  for x in `cat data/local/data/csr96_hub4/{train,test}.filelist`; do
+    gunzip -c $x
+  done | gzip -c > ${dir}/data/text/csr96_hub4.txt.gz
+  
+  # Get text from 1995 CSR-IV HUB4 corpus
+  cat data/local/data/csr95_hub4/dev95_text \
+    data/local/data/csr95_hub4/eval95_text \
+    data/local/data/csr95_hub4/train95_text | cut -d ' ' -f 2- > \
+    ${dir}/data/text/csr95_hub4.txt
+
+  # Get text from NA News supplement corpus 
+  for x in data/local/data/na_news_supp; do
+    y=`basename $x`
+    [ -f $x/corpus.gz ] && ln -sf `readlink -f $x/corpus.gz` ${dir}/data/text/${y}.txt.gz
+  done
+
+  # for reporting perplexities, we'll use the "real" dev set.
+  # note, we can't put it in ${dir}/data/text/, because then pocolm would use
+  # it as one of the data sources.
+  for x in dev96pe dev96ue eval96 eval97 eval98 eval99_1 eval99_2; do
+    cat data/$x/stm | awk '!/^;;/ {if (NF > 6) print $0}' | cut -d ' ' -f 1,7- | \
+      awk '!/IGNORE_TIME_SEGMENT_IN_SCORING/ {print $0}' | \
+      local/normalize_transcripts.pl "<NOISE>" "<SPOKEN_NOISE>" | \
+      cut -d ' ' -f 2- > ${dir}/data/${x}.txt
+  done
+fi
+
+if [ $stage -le 1 ]; then
+  mkdir -p $dir/data/work
+  if [ ! -f $dir/data/work/word_counts/.done ]; then
+    get_word_counts.py $dir/data/text $dir/data/work/word_counts
+    touch $dir/data/work/word_counts/.done
+  fi
+fi
+
+if [ $stage -le 2 ]; then
+  # decide on the vocabulary.
+
+  # NA news corpus is not clean. So better not to get vocabulary from there.
+  # for x in data/local/data/na_news/*; do
+  #   y=$dir/data/work/word_counts/`basename $x`.counts
+  #   [ -f $y ] && cat $y 
+  # done | local/lm/merge_word_counts.py 15 > $dir/data/work/na_news.wordlist_counts
+
+  cat $dir/data/work/word_counts/{train_bn96,dev}.counts | \
+    local/lm/merge_word_counts.py 2 > $dir/data/work/train_bn96.wordlist_counts
+
+  cat $dir/data/work/word_counts/csr96_hub4.counts | \
+    local/lm/merge_word_counts.py 5 > $dir/data/work/csr96_hub4.wordlist_counts
+
+  cat $dir/data/work/word_counts/csr95_hub4.counts | \
+    local/lm/merge_word_counts.py 5 > $dir/data/work/csr95_hub4.wordlist_counts
+
+  cat $dir/data/work/{train_bn96,csr96_hub4,csr95_hub4}.wordlist_counts | \
+    perl -ane 'if ($F[1] =~ m/[A-Za-z]/) { print "$F[0] $F[1]\n"; }' | \
+    local/lm/merge_word_counts.py 1 | sort -k 1,1nr > $dir/data/work/final.wordlist_counts
+
+  if [ ! -z "$vocab_size" ]; then
+    awk -v sz=$vocab_size 'BEGIN{count=-1;} 
```

Comment:
traditionally you wouldn't use up this much vertical space in an inline awk script.